tardis/montecarlo/packet_source.py

```diff
@@ -96,7 +97,7 @@ def create_blackbody_packet_nus(
         xis_prod = np.prod(xis[1:], 0)
         x = ne.evaluate("-log(xis_prod)/l")
 
-        return x * (const.k_B.cgs.value * temperature) / const.h.cgs.value
+        return (x * (const.k_B.cgs.value * temperature.to(u.K)) / const.h.cgs.value).to(u.Hz)
```

Comment:
This is not exactly how to do it. Nothing in a function with units should have `.value`, and I wouldn't force it to be in Hz, only do that where it is used in the code.